File: Dataset/Leetcode/train/111/245.java
class Solution {
    public int XXX(TreeNode root) {
        if(root==null) return 0;
        if(root.left==null && root.right==null) return 1;
        int left = Integer.MAX_VALUE;
        int right = Integer.MAX_VALUE;
        if(root.left!=null){
            left = XXX(root.left)+1;
        }
        if(root.right!=null){
            right = XXX(root.right)+1;
        }
        return (left>right)?right:left;
    }
}

File: Dataset/Leetcode/train/100/245.py
 class Solution:
    def XXX(self, p: TreeNode, q: TreeNode) -> bool:
        if not p and not q:
            return True
        elif not p or not q:
            return False
        return p.val == q.val and self.XXX(p.left, q.left) and self.XXX(p.right, q.right)

